benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/strategy/selection/RandomBreakdownSelectionStrategy.java

```diff
@@ -0,0 +1,103 @@
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Random;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com) created on 2017-06-05
+ */
+public class RandomBreakdownSelectionStrategy implements SelectionStrategy {
+
+  private static Logger logger =
+      LoggerFactory.getLogger(RandomBreakdownSelectionStrategy.class.getSimpleName());
+
+  private final MinioService minioService;
+  private final ExplorationModelDAO explorationModelDAO;
+
+  public RandomBreakdownSelectionStrategy() {
+
+    this.minioService = BenchFlowTestManagerApplication.getMinioService();
+    this.explorationModelDAO = BenchFlowTestManagerApplication.getExplorationModelDAO();
+  }
+
+  // only used for testing
+  public RandomBreakdownSelectionStrategy(MinioService minioService,
+      ExplorationModelDAO explorationModelDAO) {
+    this.minioService = minioService;
+    this.explorationModelDAO = explorationModelDAO;
+  }
+
+  @Override
+  public SelectedExperimentBundle selectNextExperiment(String testID) {
+
+    logger.info("selectNextExperiment: " + testID);
+
+    try {
+
+      // get executed exploration points
+      List<Integer> explorationPointIndices =
+          explorationModelDAO.getExplorationPointIndices(testID);
+
+      String testDefinitionYamlString =
+          IOUtils.toString(minioService.getTestDefinition(testID), StandardCharsets.UTF_8);
+
+      String deploymentDescriptorYamlString = IOUtils
+          .toString(minioService.getTestDeploymentDescriptor(testID), StandardCharsets.UTF_8);
+
+      //      JavaCompatExplorationSpace explorationSpace = explorationModelDAO.getExplorationSpace(testID);
+      ExplorationSpaceGenerator.ExplorationSpace explorationSpace =
+          ExplorationSpace.explorationSpaceFromTestYaml(testDefinitionYamlString);
+
+      // next experiment to be executed
+      int nextExplorationPoint =
+          getNextExplorationPoint(explorationSpace.size(), explorationPointIndices);
+
+      // get the experiment bundle for the given point
+      // <ExperimentDefinition, DeploymentDescriptor>
+      Tuple2<String, String> experimentBundle =
+          ExplorationSpace.generateExperimentBundle(explorationSpace, nextExplorationPoint,
+              testDefinitionYamlString, deploymentDescriptorYamlString);
+
+      return new SelectedExperimentBundle(experimentBundle._1(), // experiment definition
+          experimentBundle._2(), // deployment descriptor
+          nextExplorationPoint);
+
+
+    } catch (BenchFlowTestIDDoesNotExistException | BenchFlowDeserializationException
+        | IOException e) {
+      // should not happen
+      // TODO - handle me
+      e.printStackTrace();
+    }
+
+    return null;
+  }
+
+  private int getNextExplorationPoint(int upperBound, List<Integer> explorationPointIndices) {
+
+    Random random = new Random();
```

Comment:
@VincenzoFerme I did not find anything about this. I think also it is not possible to have such guarantees. Only way (I think) would be to generate the whole list and the shuffle it, and then checking that it is not sorted. Also I'm not sure that it is good to avoid having the same order as `one-at-a-time`. If that is what the random generator comes up with, that is also random.